install_cluster.py

```diff
@@ -0,0 +1,260 @@
+    sudo(['chmod', '444', new_key_path])
+
+
+def _generate_certs(instances_dict):
+    logger.info('Generating certificates')
+    for instances_list in instances_dict.values():
+        for instance in instances_list:
+            _generate_instance_certificate(instance)
+    copy(join(CERT_PATH, 'ca.crt'), join(CERT_PATH, 'ca.pem'))
+    os.mkdir(LOCAL_CERTS_DIR)
+    copy(CERT_PATH+'/.', LOCAL_CERTS_DIR)
+    shutil.rmtree(CERT_PATH)
+
+
+def _write_certs_to_config(config_dict, config_section, node_name):
+    cert_path = join(REMOTE_CERTS_DIR, node_name+'_cert.pem')
+    key_path = join(REMOTE_CERTS_DIR, node_name+'_key.pem')
+    ca_path = join(REMOTE_CERTS_DIR, 'ca.pem')
+
+    config_dict['prometheus']['cert_path'] = cert_path
+    config_dict['prometheus']['key_path'] = key_path
+    config_dict['prometheus']['ca_path'] = ca_path
+
+    if config_section == 'manager':
+        ssl_inputs = {}
+        cert_names = ['internal_cert_path', 'external_cert_path',
+                      'postgresql_client_cert_path']
+        key_names = ['internal_key_path', 'external_key_path',
+                     'postgresql_client_key_path']
+        ca_names = ['ca_cert_path', 'external_ca_cert_path']
+        for cert_name in cert_names:
+            ssl_inputs[cert_name] = cert_path
+        for key_name in key_names:
+            ssl_inputs[key_name] = key_path
+        for ca_name in ca_names:
+            ssl_inputs[ca_name] = ca_path
+        config_dict['ssl_inputs'] = ssl_inputs
+
+    else:
+        config_dict[config_section]['cert_path'] = cert_path
+        config_dict[config_section]['key_path'] = key_path
+        config_dict[config_section]['ca_path'] = ca_path
+
+
+def _get_postgresql_cluster_members(postgresql_instances):
+    return {
+        postgresql_instances[j].name:
+            {'ip': postgresql_instances[j].private_ip}
+        for j in range(len(postgresql_instances))}
+
+
+def _prepare_postgresql_config_files(instances_dict):
+    logger.info('Preparing PostgreSQL config files')
+    for node in instances_dict['postgresql']:
+        config_dict = get_dict_from_yaml(join(JUMP_HOST_DIR,
+                                              'postgresql_config.yaml'))
+        _write_certs_to_config(config_dict, 'postgresql_server', node.name)
+        config_dict['postgresql_server']['cluster']['nodes'] = \
+            _get_postgresql_cluster_members(instances_dict['postgresql'])
+
+        config_dict['manager']['private_ip'] = node.private_ip
+        config_dict['manager']['public_ip'] = node.public_ip
+        _create_config_file(config_dict, node.name)
+
+
+def _create_config_file(config_dict, node_name):
+    config_path = join(LOCAL_CONFIG_DIR, '{0}_config.yaml'.format(node_name))
+    with open(config_path, 'w') as config_file:
+        yaml.dump(config_dict, config_file)
+
+
+def _random_credential_generator():
+    return ''.join(random.choice(string.ascii_lowercase + string.digits)
+                   for _ in range(6))
```

Comment:
Lets make this a length of 40 or something- you could probably brute force a 6 character password (depending on the situation) with your phone in minutes or hours (depending whether it was online or offline).

Reply:
Switched to 40.